src/util/report.py

```diff
@@ -105,6 +106,10 @@ def add_to_migration_reports(message, category="Other", format="text"):
         else:
             raw = True
     migration_reports.setdefault(category, []).append((message, raw))
+    global migration_reports_length  # noqa: PLW0603
+    migration_reports_length += len(message)
```

Comment:
Do we really needs a global?
This is roughly equivalent to
```suggestion
    migration_reports_length = len(repr(migration_reports))
```

Reply:
It's just much slower. In fact, if we don't want to keep the global variable we can compute the sum every time. Wdyt?

```python
from timeit import timeit


big_messages = {i: "a"*10**i for i in range(10)}
many_messages = {i: "a"*i for i in range(1000)}

def _repr_sol(d):
    return len(repr(d))


def _total_len(d):
    return sum(len(v) for v in d.values())


print(timeit(lambda: _repr_sol(big_messages), number=10))       # 14.92
print(timeit(lambda: _total_len(big_messages), number=10))      # 1.8e-05
print(timeit(lambda: _repr_sol(many_messages), number=10))      # 0.006
print(timeit(lambda: _total_len(many_messages), number=10))     # 0.0004
```

Reply:
Yes
```suggestion
    migration_reports_length = sum(len(m) for (m, r) in migration_reports.values())
```